packages/vite/src/node/plugins/resolve.ts

```diff
@@ -487,180 +486,167 @@ function tryFsResolve(
   options: InternalResolveOptions,
   tryIndex = true,
   targetWeb = true,
+  skipPackageJson = false,
 ): string | undefined {
-  const { file, postfix } = splitFileAndPostfix(fsPath)
-
-  // Dependencies like es5-ext use `#` in their paths. We don't support `#` in user
-  // source code so we only need to perform the check for dependencies.
-  const tryUnsplitted = fsPath.includes('#') && fsPath.includes('node_modules')
```

Comment:
If there was a `#` before the query, before we were checking for `tryUnsplitted` branches with the full path including the query and these were all redundant checks.